components/Home/sidebar/HomeSidebarBountiesSection.tsx

```diff
@@ -0,0 +1,138 @@
+import { css } from "aphrodite";
+import { emptyFncWithMsg } from "~/config/utils/nullchecks";
+import {
+  fetchOpenBounties,
+  SimpleBounty,
+} from "~/components/Bounty/api/fetchOpenBounties";
+import { formatBountyAmount } from "~/config/types/bounty";
+import { getFEUnifiedDocType } from "~/config/utils/getUnifiedDocType";
+import { Fragment, ReactElement, useEffect, useState } from "react";
+import { SideColumnTitle } from "~/components/Typography";
+import { styles } from "./styles/HomeRightSidebarStyles";
+import BountiesSidebarItem from "./sidebar_items/BountiesSidebarItem";
+import colors from "~/config/themes/colors";
+import HubEntryPlaceholder from "~/components/Placeholders/HubEntryPlaceholder";
+import Link from "next/link";
+import ReactPlaceholder from "react-placeholder/lib";
+
+type PaginationInfo = { isFetching: boolean; page?: number };
+
+const useEffectFetchOpenBounties = ({
+  paginationInfo,
+  setOpenBounties,
+  setPaginationInfo,
+}: {
+  paginationInfo: PaginationInfo;
+  setOpenBounties: (bounties: any) => void;
+  setPaginationInfo: (paginationInfo: PaginationInfo) => void;
+}): void => {
+  const { isFetching, page } = paginationInfo;
+  useEffect((): void => {
+    if (isFetching) {
+      fetchOpenBounties({
+        onSuccess: (bounties: SimpleBounty) => {
+          // TODO: calvinhlee deal with page when supported by BE
+          setPaginationInfo({ isFetching: false, page });
+          setOpenBounties(bounties);
+        },
+        onError: emptyFncWithMsg,
+      });
+    }
+  }, [isFetching]);
+};
+
+export default function HomeSidebarBountiesSection(): ReactElement {
+  const [paginationInfo, setPaginationInfo] = useState<PaginationInfo>({
+    isFetching: true,
+    page: 1,
+  });
+  const [openBounties, setOpenBounties] = useState<SimpleBounty[]>([]);
+
+  useEffectFetchOpenBounties({
+    paginationInfo,
+    setOpenBounties,
+    setPaginationInfo,
+  });
+
+  const { isFetching, page = 1 } = paginationInfo;
+  const isReadyToRender = !isFetching && page > 0;
+  const _isLoadingMore = !isFetching && page !== 1;
+
+  const bountyItems = openBounties?.map(
```

Comment:
Most of the logic in this function is handled by the `Bounty` class.

documents, amounts, expiration date formatting, ...